### PR TITLE
INTMDB-299: Added snapshot export jobs support

### PR DESCRIPTION
**Pull request by abner-dou**
## Description

Added support for `CloudProviderSnapshotExportJobs`

Link to any related issue(s): [INTMDB-299](https://jira.mongodb.org/browse/INTMDB-299)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: mongodbatlas/cloud_provider_snapshot_export_jobs.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+type CloudProviderSnapshotExportJobsServiceOp service
+
+// List Retrieve all the export jobs for the specified project.
+//
+// See more: https://docs.atlas.mongodb.com/reference/api/cloud-backup/export/get-all-export-jobs/
+func (c CloudProviderSnapshotExportJobsServiceOp) List(ctx context.Context, projectID, clusterName string, options *ListOptions) (*CloudProviderSnapshotExportJobs, *Response, error) {
+	if projectID == "" {
+		return nil, nil, NewArgError("projectID", "must be set")
+	}
+
```
Comment (gssbzn) [inline mongodbatlas/cloud_provider_snapshot_export_jobs.go:46]:
```suggestion
	if clusterName == "" {
		return nil, nil, NewArgError("clusterName", "must be set")
	}
```

File: mongodbatlas/cloud_provider_snapshot_export_jobs.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+// Get Allows you to retrieve one export job specified by the bucket ID.
+//
+// See more: https://docs.atlas.mongodb.com/reference/api/cloud-backup/export/get-one-export-job/
+func (c CloudProviderSnapshotExportJobsServiceOp) Get(ctx context.Context, projectID, clusterName, bucketID string) (*CloudProviderSnapshotExportJob, *Response, error) {
+	if projectID == "" {
+		return nil, nil, NewArgError("projectID", "must be set")
+	}
+	if bucketID == "" {
```
Comment (gssbzn) [inline mongodbatlas/cloud_provider_snapshot_export_jobs.go:79]:
```suggestion
	if clusterName == "" {
		return nil, nil, NewArgError("clusterName", "must be set")
	}
	if bucketID == "" {
```

File: mongodbatlas/cloud_provider_snapshot_export_jobs.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+// Create Allows you to grant Atlas access to the specified export job for exporting backup snapshots.
+//
+// See more: https://docs.atlas.mongodb.com/reference/api/cloud-backup/export/create-one-export-job/
+func (c CloudProviderSnapshotExportJobsServiceOp) Create(ctx context.Context, projectID, clusterName string, bucket *CloudProviderSnapshotExportJob) (*CloudProviderSnapshotExportJob, *Response, error) {
+	if projectID == "" {
+		return nil, nil, NewArgError("projectID", "must be set")
+	}
+
```
Comment (gssbzn) [inline mongodbatlas/cloud_provider_snapshot_export_jobs.go:106]:
```suggestion
	if clusterName == "" {
		return nil, nil, NewArgError("clusterName", "must be set")
	}
```

File: mongodbatlas/cloud_provider_snapshot_export_jobs.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+	ErrMsg         string                                      `json:"errMsg,omitempty"`         // Error message, only if the export job failed.
+	ExportBucketID string                                      `json:"exportBucketId,omitempty"` // Unique identifier of the bucket.
+	ExportStatus   *CloudProviderSnapshotExportJobStatus       `json:"exportStatus,omitempty"`   // Returned for replica set only. Status of the export job.
+	FinishedAt     string                                      `json:"finishedAt,omitempty"`     // Timestamp in ISO 8601 date and time format in UTC when the export job completes.
+	Prefix         string                                      `json:"prefix,omitempty"`         // Full path on the cloud provider bucket to the folder where the snapshot is exported. The path is in the following format: /exported_snapshots/{ORG-NAME}/{PROJECT-NAME}/{CLUSTER-NAME}/{SNAPSHOT-INITIATION-DATE}/{TIMESTAMP}
+	SnapshotID     string                                      `json:"snapshotID,omitempty"`     // Unique identifier of the snapshot.
```
Comment (gssbzn) [inline mongodbatlas/cloud_provider_snapshot_export_jobs.go:158]:
```suggestion
	SnapshotID     string                                      `json:"snapshotId,omitempty"`     // Unique identifier of the snapshot.
```

**Review by gssbzn — CHANGES_REQUESTED** [review]
Couple of small things to fix

File: mongodbatlas/cloud_provider_snapshot_export_jobs.go (hunk 2 above)
Comment (abner-dou) [reply mongodbatlas/cloud_provider_snapshot_export_jobs.go:86]: changed :)

File: mongodbatlas/cloud_provider_snapshot_export_jobs.go (hunk 3 above)
Comment (abner-dou) [reply mongodbatlas/cloud_provider_snapshot_export_jobs.go:116]: changed :)

File: mongodbatlas/cloud_provider_snapshot_export_jobs.go (hunk 4 above)
Comment (abner-dou) [reply mongodbatlas/cloud_provider_snapshot_export_jobs.go]: changed :)

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**